### ISSUE TITLE
Let's get your email address for Drummer

### ISSUE THREAD

**Issue by scripting** [scripting/drummerSupport#177]
Here's a new little app to help the transition from the old Drummer to the new. 

http://scripting.com/code/readtweet/

### How to

0. If you haven't previously created an account on [drummer.land](https://drummer.land/), do so now.
1. In Twitter create a new tweet containing only the text of your email address that you use to log on to drummer.land.
2. Copy the URL of the tweet to the clipboard.
3. Open the [web page](http://scripting.com/code/readtweet/) for the app, and paste the URL of the tweet in the edit box. Click Go.
4. Our server will read the tweet, and if it correctly finds the email address, will record it in our database.
5. You can then delete the tweet. It isn't needed any longer.
6. If there's an error, it will show you what the error is. ;-)
7. If you have questions post them here. 

### What then?

Assuming there are some items in the database, I will manually move the outlines from the old server to an account on drummer.land, using the email address to identify each user. I'll post a note here when that's done. Then you can log on to drummer.land and hopefully verify that your files have arrived.

**Comment by akaKenSmith** [comment]
This went smoothly . . .

**Comment by scripting** [comment]
BTW, this isn't exactly a test. If you want your Drummer files moved to the new server, you _have_ to do this. 

And the longer you wait, the longer you risk the Twitter API is going to break. The time to do this is the first chance you get. 

My apologies for not making this clear.

**Comment by allthisness** [comment]
Mmm, not sure what went wrong…
![6083EA7F-9DA5-4241-955C-DE09AC0A314C](https://user-images.githubusercontent.com/121541188/218806207-c3db9154-9f8b-4bb0-a97e-8db9fabc2e98.jpeg)

**Comment by scripting** [comment]
@allthisness -- are you on a corporate network that limits what you can do? any plugins? i just tested the app and it's working.

**Comment by allthisness** [comment]
Thank you. I noticed that sharing the URL out of Twitter added a bunch of crud to the end which I got rid of. Success! Thank you 🙏

**Comment by scripting** [comment]
@allthisness -- good work!

**Comment by grantdunham** [comment]
> Thank you. I noticed that sharing the URL out of Twitter added a bunch of crud to the end which I got rid of. Success! Thank you 🙏

The same here. It would not work until I deleted the extra crud at the end of the URL. But works well once that's done.

**Comment by jackbaty** [comment]
I needed to turn off "protected tweets" but once I did that, it seemed to work. It found my address correctly and so I clicked "OK".

**Comment by scripting** [comment]
@jackbaty -- that's the idea. the url parser on the server is very low rent. ;-)

**Comment by scripting** [comment]
I've moved four users.

- "KenSmith", "ksmith3123@gmail.com"
- "mistersugar", "anton@zuiker.com"
- "ScottHansonDE", "papascott@gmail.com"
- "davewiner", "dave.winer@gmail.com"

Can you go to your account on drummer.land and verify that what you see there are your files from the previous server, plus whatever files you've been working on here. 

Once these are confirmed, I'll do the rest. And periodically do batches.

**Comment by mistersugar** [comment]
First attempt not successful.

I went to drummer.land, entered my email, saw a dialog saying to check my email.

I checked my email and clicked on the link, which opened Safari but immediately gave me a dialog about not having an account.

Screenshot attached.

No errors in the console.

<img width="561" alt="Screenshot 2023-02-14 at 1 56 16 PM" src="https://user-images.githubusercontent.com/3846301/218831953-323639ba-d1b7-4560-b8f4-070a86df7e4e.png">

**Comment by scripting** [comment]
@mistersugar -- please a little more info. were you able to log on before? did you create an account previously? this app does not create the account for you.

**Comment by mistersugar** [comment]
Update to above: I first clicked on "Log on"

I've just tried again by clicking on "Sign up" and I am in,

I will check my files now.

**Comment by mistersugar** [comment]
> 



> @mistersugar -- please a little more info. were you able to log on before? did you create an account previously? this app does not create the account for you.

That was my mistake in thinking I had an account on this.

I do see my files, which match the files in my recent backup download.

I have opened and closed a few.

Looks good on my end.

**Comment by akaKenSmith** [comment]
I was able to go to my account on drummer.land and verify that what I see there are my files from the previous server, plus the file I've been working on since we moved to this new server.

**Comment by scripting** [comment]
I've moved the next group of accounts. 

- "AndySylvester99", "sylvester.andy@gmail.com"
- "PDXProtestNews", "andy@andysylvester.com"
- "allthisness", "nick.reynolds@gmail.com"
- "gwthompson", "gwthompson@mac.com"
- "gdunham", "grant.dunham@gmail.com"
- "jackbaty", "jack@baty.net"
- "hungrygarden", "puckett.brian@gmail.com"

**Comment by allthisness** [comment]
Bingo! I’m successfully in! 😀🙏

**Comment by troutcolor** [comment]
I've now gone through the process. Thanks.

**Comment by jackbaty** [comment]
Looks like I'm in and everything moved just fine. Nice!

One odd thing I'm seeing is that there is an open "Change notes" tab that I am unable to close. Clicking on it switches the active tab, but the contents don't change. There is no close icon (x) and if it's selected and I use "File->close" it closes the previously-selected tab. Here's a quick screen recording I made. Hope it helps. I'm using Arc (Chrome) and I've deleted the cache and reloaded but no change.

![20230214-New Drummer](https://user-images.githubusercontent.com/970/218867694-3a872ee8-0d3c-4462-a332-0cd48a0a2e89.gif)

**Comment by scripting** [comment]
@jackbaty -- i know what the bug is and how to fix it. i'll post here when it's fixed, and if you could try it then, much appreciated.

**Comment by scripting** [comment]
@troutcolor -- i've moved your files to the new account.

**Comment by troutcolor** [comment]
> @troutcolor -- i've moved your files to the new account.

Hi Dave,
Thanks, I seem to have run int othe case sensitive email problem here. I used all lower case in twitter, so I get 

>The user "johnjohnston@mac.com" doesn't have an account on this server.

error if I use that.
If I use Johnjohnston I can log on but don't see the old outlines.

Cheers

John

**Comment by scripting** [comment]
@troutcolor -- do the form with the email address you want to use and i'll run it through again.

**Comment by rtanglao** [comment]
* i'e gone through the process thanks! worked great! please put me in a future batch: rolandt AT gmail.com twitter.com/rtanglao

**Comment by allthisness** [comment]
I’m guessing Drummer blogs have yet to got working?
![75BA677D-8A46-4DD0-A09B-05320B7D625D](https://user-images.githubusercontent.com/121541188/218886061-68aa7c98-7a74-4077-a778-5fbb0a020c03.jpeg)

**Comment by mistersugar** [comment]
> @jackbaty -- i know what the bug is and how to fix it. i'll post here when it's fixed, and if you could try it then, much appreciated.

Perhaps the tips [here](http://scripting.com/drummer/blog/2021/11/24/142717.html?title=tabWithNoCloseBox) will solve this?

**Comment by jackbaty** [comment]
> Perhaps the tips [here](http://scripting.com/drummer/blog/2021/11/24/142717.html?title=tabWithNoCloseBox) will solve this?

Well look at that :). Thanks, but I think I should wait to confirm that Dave's pending fix works first.

**Comment by scripting** [comment]
@jackbaty -- i forgot i did that, you can go ahead and use it. i can test it sufficiently here.

**Comment by jackbaty** [comment]
@scripting, the instructions on that page worked fine for closing that tab, thanks. (thanks for the tip, @mistersugar).

**Comment by spkaluzny** [comment]
When I enter my tweet [URL](https://twitter.com/spkaluzny/status/1625695046266937345) I get error box popping up that says:
> There was an error communicating with the server.

**Comment by troutcolor** [comment]
> @troutcolor -- do the form with the email address you want to use and i'll run it through again.

@scripting Thanks, I've doen it again all lowercase.

cheers

john

**Comment by scripting** [comment]
I'm not going to post the list of names from this point on, just that i'm uptodate with all the people who have asked that their files be updated. 

The next thing is to redirect from drummer.scripting.com to drummer.land. 

BTW, once the dust settles, we have to get the blogs hooked up to the new server. But I have to turn my attention back to FeedLand now.

**Comment by am1t** [comment]
This worked great for me. It's nice to see all the posts and files back. Thank you, Dave!

**Comment by dmatchett** [comment]
Cross-posting from #178, because there's so much more action here: the Safari bookmarklet invokes the Drummer notes tab, but doesn't create a new entry with the link. My number one hope is to have this working again: it's what made Drummer the fabulous tool it was. Hope you have time to tweak the bookmarklet! (I'm not a programmer, perhaps you can tell...) Thanks a million.

**Comment by jgrisham** [comment]
Is this migration service still available? I can't seem to get the form to accept a Tweet URL.
> _(Yea, I'm a little behind the curve, here: I mostly still use a combination of [Little Outliner](http://littleoutliner.com/), [Opal](https://a-sharp.com/opal/), and [wikidPad](https://wikidpad.sourceforge.net/); didn't catch up on Drummer news from 2023 until about a week ago. This is mostly a curiosity to me personally, as I can always restore from backup, but it might affect other infrequent users as well.)_

<hr />
<details><summary>(show/hide details)
</summary>

<p />

The page referenced in the first post above, titled _[Get Email address from Twitter account](http://scripting.com/code/readtweet/)_, is still live at:
> http://scripting.com/code/readtweet/

... but when submitting a tweet URL, an error occurs, stating:
> There was an error communicating with the server.

Per the Javascript console, a HTTP 404 status code is received for:

> `http://readtweet.scripting.com/getemailaddressfromtweet?urltweet=https%3A%2F%2Ftwitter.com%2Fexampleuser%2Fstatus%2F1664437816858923009`

With the specific console error message:
> "GET http://scripting.com/code/getemailaddressfromtweet 404 (Not Found)"
<details><summary>(screenshot)
</summary>

![image](https://github.com/scripting/drummerSupport/assets/1781752/350fc6f7-eb84-41fa-94ad-f10ac7557f11)

</details>

<hr />
Environment:

- I received identical results on macOS 10.15 with current versions of both Opera and Edge, and also on iOS using Safari.
- Those devices are on two completely different USA-based networks.

<hr />

</details>

Has anyone successfully completed the above process recently?

<hr />

**2023-08-08 update**: there doesn't appear to have been any change in the past 70 days.

Tl;dr: The API URL ( http://scripting.com/code/getemailaddressfromtweet ) called by the form + JavaScript on the [above page](http://scripting.com/code/readtweet/) returns with a HTTP status code of 404; _(see the hidden section above for further details)_.

Are there any other ways _(besides bothering DW)_ of linking a 'legacy' Drummer account with a new-style _(i.e. e-mail address-based)_ one?
